instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -228,4 +120,24 @@ public OpenTelemetry openTelemetry() {
       return OpenTelemetry.noop();
     }
   }
+
+  public static class OpenTelemetrySdkComponentLoader implements ComponentLoader {
+    private final ApplicationContext applicationContext;
+
+    private final SpiHelper spiHelper =
+        SpiHelper.create(OpenTelemetrySdkComponentLoader.class.getClassLoader());
+
+    public OpenTelemetrySdkComponentLoader(ApplicationContext applicationContext) {
+      this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public <T> Iterable<T> load(Class<T> spiClass) {
+      List<T> spi = spiHelper.load(spiClass);
+      List<T> beans =
+          applicationContext.getBeanProvider(spiClass).stream().collect(Collectors.toList());
```

Comment:
From the [Spring doc](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/ObjectProvider.html#orderedStream()):

>default [Stream](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/stream/Stream.html)<[T](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/ObjectProvider.html)> stream()
Return a sequential [Stream](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/stream/Stream.html) over all matching object instances, **without specific ordering guarantees (but typically in registration order)**.

Could you please complete the test with a second `AutoConfigurationCustomizerProvider` bean having a different order priority than the first one and creating another `ResourceCustomizer` also defining the `keyFromResourceCustomizer` attribute?

To make it work the code may be: `applicationContext.getBeanProvider(spiClass).orderedStream().collect(Collectors.toList());`

Reply:
yes, good idea

Reply:
Thinking about it, I changed my mind:

The SDK has it's own [Ordered](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/Ordered.java) interface, which I think should be used.
If we support both the spring `@Order` annotation, this might be confusing - and we cannot take `Ordered` away.

Reply:
If we allow users to define `AutoConfigurationCustomizerProvider` as beans then they can expect that Spring Boot annotations such as `org.springframework.core.annotation.Order` will work, because they use the Spring way with a the Spring OpenTelemetry starter they have added to their project.

Reply:
> The SDK has it's own [Ordered](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/Ordered.java) interface, which I think should be used.

Could you provide an example on how to use the `Ordered` interface with an `AutoConfigurationCustomizerProvider` bean?

Reply:
You can override the `order` method to return a value, like here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/JarServiceNameDetector.java#L131

Reply:
I'm okay supporting both styles, you made a good argument